dev-tools/mage/settings.go

```diff
@@ -561,13 +562,80 @@ var (
 // import path and the current directory's import path.
 func GetProjectRepoInfo() (*ProjectRepoInfo, error) {
 	repoInfoOnce.Do(func() {
-		repoInfoValue, repoInfoErr = getProjectRepoInfo()
+		if isUnderGOPATH() {
+			repoInfoValue, repoInfoErr = getProjectRepoInfoUnderGopath()
+		} else {
+			repoInfoValue, repoInfoErr = getProjectRepoInfoWithModules()
+		}
 	})
 
 	return repoInfoValue, repoInfoErr
 }
 
-func getProjectRepoInfo() (*ProjectRepoInfo, error) {
+func isUnderGOPATH() bool {
+	rel, err := filepath.Rel(build.Default.GOPATH, CWD())
+	if err != nil {
+		return false
+	}
+
+	if strings.Contains(rel, "..") {
+		return false
+	}
+
+	return true
+}
+
+func getProjectRepoInfoWithModules() (*ProjectRepoInfo, error) {
+	var (
+		cwd            = CWD()
+		rootImportPath string
+		rootDir        string
+		subDir         string
+	)
+
+	possibleRoot := cwd
+	var errs []string
+	for {
+		isRoot, err := isRepoRoot(possibleRoot)
+		if err != nil {
+			errs = append(errs, err.Error())
+		}
+
+		if isRoot {
+			rootDir = possibleRoot
+			break
+		}
+
+		subDir = path.Base(possibleRoot)
```

Comment:
If you are multiple levels deep in project relative to the root then I don't think this value of subdir will be correct. I think it needs to come from `filepath.Rel` after you know where the root is located.